File: app/src/main/java/com/example/bottlerocketstudioscodingtest/Model/Store.java
package com.example.bottlerocketstudioscodingtest.Model;

import android.os.Parcel;
import android.os.Parcelable;

import com.google.gson.annotations.SerializedName;

public class Store implements Parcelable {
    @SerializedName("address")
    private String address;
    @SerializedName("city")
    private String city;
    @SerializedName("name")
    private String name;
    @SerializedName("storeLogoURL")
    private String storeLogoURL;
    @SerializedName("phone")
    private String phone;
    @SerializedName("state")
    private String state;
    @SerializedName("zipcode")
    private String zipcode;

    public Store(String address, String city, String name, String storeLogoURL, String phone,
                 String state, String zipcode){
        this.address = address;
        this.city = city;
        this.name = name;
        this.storeLogoURL = storeLogoURL;
        this.phone = phone;
        this.state = state;
        this.zipcode = zipcode;
    }

    protected Store(Parcel in) {
        address = in.readString();
        city = in.readString();
        name = in.readString();
        storeLogoURL = in.readString();
        phone = in.readString();
        state = in.readString();
        zipcode = in.readString();
    }

    public static final Creator<Store> CREATOR = new Creator<Store>() {
        @Override
        public Store createFromParcel(Parcel in) {
            return new Store(in);
        }

        @Override
        public Store[] newArray(int size) {
            return new Store[size];
        }
    };

    public String getAddress() {
        return address;
    }

    public void setAddress(String address) {
        this.address = address;
    }

    public String getCity() {
        return city;
    }

    public void setCity(String city) {
        this.city = city;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getStoreLogoURL() {
        return storeLogoURL;
    }

    public void setStoreLogoURL(String storeLogoURL) {
        this.storeLogoURL = storeLogoURL;
    }

    public String getPhone() {
        return phone;
    }

    public void setPhone(String phone) {
        this.phone = phone;
    }

    public String getState() {
        return state;
    }

    public void setState(String state) {
        this.state = state;
    }

    public String getZipcode() {
        return zipcode;
    }

    public void setZipcode(String zipcode) {
        this.zipcode = zipcode;
    }

    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public void writeToParcel(Parcel parcel, int i) {
        parcel.writeString(address);
        parcel.writeString(city);
        parcel.writeString(name);
        parcel.writeString(storeLogoURL);
        parcel.writeString(phone);
        parcel.writeString(state);
        parcel.writeString(zipcode);
    }
}

File: app/src/main/java/com/example/bottlerocketstudioscodingtestkotlin/Model/Store.kt
package com.example.bottlerocketstudioscodingtestkotlin.Model

import android.os.Parcel
import android.os.Parcelable

import com.google.gson.annotations.SerializedName

class Store : Parcelable {
    @SerializedName("address")
    var address: String? = null
    @SerializedName("city")
    var city: String? = null
    @SerializedName("name")
    var name: String? = null
    @SerializedName("storeLogoURL")
    var storeLogoURL: String? = null
    @SerializedName("phone")
    var phone: String? = null
    @SerializedName("state")
    var state: String? = null
    @SerializedName("zipcode")
    var zipcode: String? = null
    @SerializedName("latitude")
    var latitude: String? = null
    @SerializedName("longitude")
    var longtitude: String? = null
    @SerializedName("storeID")
    var storeID: String? = null

    constructor(address: String, city: String, name: String, storeLogoURL: String, phone: String,
                state: String, zipcode: String, latitude: String, longitude: String, storeID: String) {
        this.address = address
        this.city = city
        this.name = name
        this.storeLogoURL = storeLogoURL
        this.phone = phone
        this.state = state
        this.zipcode = zipcode
        this.latitude = latitude
        this.longtitude = longitude
        this.storeID = storeID
    }

    protected constructor(`in`: Parcel) {
        address = `in`.readString()
        city = `in`.readString()
        name = `in`.readString()
        storeLogoURL = `in`.readString()
        phone = `in`.readString()
        state = `in`.readString()
        zipcode = `in`.readString()
        latitude = `in`.readString()
        longtitude = `in`.readString()
        storeID = `in`.readString()
    }

    override fun describeContents(): Int {
        return 0
    }

    override fun writeToParcel(parcel: Parcel, i: Int) {
        parcel.writeString(address)
        parcel.writeString(city)
        parcel.writeString(name)
        parcel.writeString(storeLogoURL)
        parcel.writeString(phone)
        parcel.writeString(state)
        parcel.writeString(zipcode)
        parcel.writeString(latitude)
        parcel.writeString(longtitude)
        parcel.writeString(storeID)
    }

    /*companion object {

        val CREATOR: Parcelable.Creator<Store> = object : Parcelable.Creator<Store> {
            override fun createFromParcel(`in`: Parcel): Store {
                return Store(`in`)
            }

            override fun newArray(size: Int): Array<Store> {
                return arrayOfNulls(size)
            }
        }
    }*/

    companion object CREATOR : Parcelable.Creator<Store> {
        override fun createFromParcel(parcel: Parcel): Store {
            return Store(parcel)
        }

        override fun newArray(size: Int): Array<Store?> {
            return arrayOfNulls(size)
        }
    }
}
